sqlobject/src/main/java/org/jdbi/v3/sqlobject/config/internal/RegisterCollectorFactoryImpl.java

```diff
@@ -32,12 +31,11 @@ public void configureForMethod(ConfigRegistry registry, Annotation annotation, C
     public void configureForType(ConfigRegistry registry, Annotation annotation, Class<?> sqlObjectType) {
         RegisterCollectorFactory registerCollectorFactory = (RegisterCollectorFactory) annotation;
         JdbiCollectors collectors = registry.get(JdbiCollectors.class);
-        for (Class<? extends CollectorFactory> type : registerCollectorFactory.value()) {
-            try {
-                collectors.register(type.newInstance());
-            } catch (InstantiationException | IllegalAccessException e) {
-                throw new IllegalStateException("Unable to instantiate container factory", e);
-            }
+        Class<? extends CollectorFactory> type = registerCollectorFactory.value();
+        try {
+            collectors.register(type.newInstance());
+        } catch (InstantiationException | IllegalAccessException e) {
```

Comment:
Consider catching the new-ish `ReflectiveOperationException` which groups all these miscellaneous problems nicely.